internal/app/machined/pkg/controllers/k8s/manifest_test.go

```diff
@@ -171,9 +178,7 @@ func (suite *ManifestSuite) TestReconcileKubeProxyExtraArgs() {
 	rootSecrets := secrets.NewRoot(secrets.RootKubernetesID)
 	manifestConfig := config.NewK8sManifests()
 	spec := defaultManifestSpec
-	spec.ProxyExtraArgs = map[string]string{
-		"bind-address": "\"::\"",
-	}
+	spec.ProxyArgs = append(spec.ProxyArgs, "--bind-address=\"::\"")
```

Comment:
this test becomes kinda weird after I changed the logic. Not sure what's the right way to fix it.

Reply:
yep, this was testing mostly JSON escaping, an I think it's wrong now probably

Reply:
yeah... Let me check

Reply:
so in terms of escaping I think it works fine:

```yaml
    proxy:
        image: k8s.gcr.io/kube-proxy:v1.21.3 # The container image used in the kube-proxy manifest.
        extraArgs:
          bind-address: "\"::\""
```

Turned into:

```
        - /usr/local/bin/kube-proxy
        - --cluster-cidr=10.244.0.0/16
        - --proxy-mode=iptables
        - --bind-address="::"
        - --hostname-override=$(NODE_NAME)
        - --kubeconfig=/etc/kubernetes/kubeconfig
        - --conntrack-max-per-core=0
```

Reply:
having `| json` shouldn't hurt in the template, as YAML is a subset of JSON, so we can put array values as JSON formatted strings, and this avoids any kind of YAML weirdness that might appear there

Reply:
not sure if the test was completely correct though, but there was something up with `:` iirc, so that it treats part of the list as a map